### PR TITLE
Warn instead of raising exceptions in inf-check

### PR DESCRIPTION
**Pull request by zhu-han**

### CODE REVIEW

File: icefall/hooks.py [hunk 1]
```diff
@@ -40,8 +40,8 @@ def register_inf_check_hooks(model: nn.Module) -> None:
         def forward_hook(_module, _input, _output, _name=name):
             if isinstance(_output, Tensor):
                 if not torch.isfinite(_output.to(torch.float32).sum()):
-                    raise ValueError(
-                        f"The sum of {_name}.output is not finite: {_output}"
+                    logging.warning(
```
Comment (csukuangfj) [inline icefall/hooks.py:43]: Will the training continue with inf loss and inf grad?

Comment (zhu-han) [reply icefall/hooks.py:43]: Yes, like the backward code, it will just warn instead of printing the tensor and raising an exception.

Comment (csukuangfj) [reply icefall/hooks.py:43]: In that case, will it propagate the inf grad to model parameters and result in inf parameters?

Comment (csukuangfj) [reply icefall/hooks.py:43]: I see. Thanks!

**Comment by csukuangfj** [comment]
Could you fix the style issues?

**Review by csukuangfj — APPROVED** [review]
Thanks!
